psi4/driver/driver_cbs.py

```diff
@@ -1497,11 +1226,574 @@ def _cbs_gufunc(func, total_method_name, **kwargs):
+            required.append((lvl['f_wfn'], lvl['f_basis'], lvl['f_options']))
+
+    tables += headfmt.format('', 'Method', '/', 'Basis', 'Rqd', 'Energy [Eh]', 'Variable')
+    tables += delimit
+    for job in TROVE:
+        star = ''
+        for mc in required:
+            if (job['f_wfn'], job['f_basis'], job['f_options']) == mc:
+                star = '*'
+        tables += linefmt.format('', job['f_wfn'], '/', job['f_basis'] + " + options" * bool(job['f_options']), star,
+                                 job['f_energy'], VARH[job['f_wfn']][job['f_wfn']])
+    tables += delimit
+
+    tables += blckfmt.format('Stages')
+    tables += delimit
+    tables += headfmt.format('Stage', 'Method', '/', 'Basis', 'Wt', 'Energy [Eh]', 'Scheme')
+    tables += delimit
+    for stage in GRAND_NEED:
+        tables += linefmt.format(stage['d_stage'], stage['d_wfn'], '/', stage['d_basis'], stage['d_coef'],
+                                 stage['d_energy'], stage['d_scheme'])
+    tables += delimit
+
+    tables += blckfmt.format('CBS')
+    tables += delimit
+    tables += headfmt.format('Stage', 'Method', '/', 'Basis', '', 'Energy [Eh]', 'Scheme')
+    tables += delimit
+    tables += linefmt.format(GRAND_NEED[0]['d_stage'], GRAND_NEED[0]['d_wfn'], '/', GRAND_NEED[0]['d_basis'], '',
+                             GRAND_NEED[0]['d_energy'], GRAND_NEED[0]['d_scheme'])
+
+    if len(metadata) > 1:
+        dc = 1
+        for delta in metadata[1:]:
+            mtdstr = GRAND_NEED[dc]['d_wfn']
+            if dc != 1:
+                mtdstr += ' - ' + GRAND_NEED[dc + 1]['d_wfn']
+            tables += linefmt.format(GRAND_NEED[dc]['d_stage'], mtdstr, '/', GRAND_NEED[dc]['d_basis'], '',
+                                     GRAND_NEED[dc]['d_energy'] - GRAND_NEED[dc + 1]['d_energy'],
+                                     GRAND_NEED[dc]['d_scheme'])
+            dc += 2
+
+    tables += linefmt.format('total', 'CBS', '', '', '', finalenergy, '')
+    tables += delimit
+
+    return tables
+
+
+class CompositeComputer(BaseComputer):
+
+    molecule: Any
+    basis: str = "(auto)"
+    method: str = "(auto)"
+    driver: DriverEnum
+    keywords: Dict[str, Any] = {}
+    metadata: Any
+    metameta: Dict[str, Any] = {}
+
+    verbose: int = 1
+
+    # List of model chemistries with extrapolation scheme applied. Can reconstruct CBS. Keys are d_fields. Formerly GRAND_NEED.
+    cbsrec: List[Dict[str, Any]] = []
+
+    # Maximal list of model chemistries extractable from running `compute_list`. Keys are _f_fields. Formerly JOBS_EXT.
+    trove: List[Dict[str, Any]] = []
+
+    # Minimal (enlightened) list of jobs to run to satisfy full CBS. Keys are _f_fields. Formerly JOBS.
+    compute_list: List[Dict[str, Any]] = []
+
+    # One-to-One list of AtomicComputer-s corresponding to `compute_list`.
+    task_list: List[AtomicComputer] = []
+
+    # One-to-One list of QCSchema corresponding to `task_list`.
+    results_list: List[Any] = []
```

Comment:
The issue of the default `[]` and `{}` being a class variable as opposed to an instance variable applies here too. Should be `Field(default_factory=list/dict)`

Reply:
Will do, when I get to `pydantic.Field`ifying this class.